main.go

```diff
@@ -6,52 +6,65 @@ import (
 	"path/filepath"
 	"flag"
 
-	"github.com/docker/docker/builder/dockerignore"
-	"github.com/docker/docker/pkg/fileutils"
+	"./dockerignore"
+	"./fileutils"
 )
 
+// Declare the array of type String
 var excludes []string
 
 func main() {
+	// Package flag implements command-line flag parsing.
+	// Parse() parses the command-line flags from os.Args[1:]. Must be called after all flags are defined and before flags are accessed by the program.
 	flag.Parse()
-	root := flag.Arg(0)
-	if len(root) == 0 {
-		fmt.Printf ("Pass in file path to examine.\n")
-		panic("Missing main argument")
-	}
+	// Check the number of command-line arguments.
+	// It should be 2: the .go file itslef and the path to inspect.
+	if len(os.Args) != 2 {
+		panic("Please provide a valid path to be inspected.")
+	}	
 
```

Comment:
The root variable was assigned before checking if the source will even provide a value! :-)
I will modify as it was.